dipy/reconst/ivim.py

```diff
@@ -147,7 +146,8 @@ def nlls_fit(data, gtab, jac=False):
     bvals = gtab.bvals
     ivim_params = np.empty((flat_data.shape[0], 4))
     for vox in range(flat_data.shape[0]):
-        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox])
+        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox],
```

Comment:
Oh I see you added a guess but your D and D* guesses are still the same here which may be why they're getting swapped.